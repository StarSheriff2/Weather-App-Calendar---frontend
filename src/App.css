:root {
  --light-gray: #f7f7f7;
  --speed: 500ms;
}

label {
  display: block;
  margin-top: 0.6rem;
}

.login-form,
.signup-form {
  max-width: 350px;
  margin: auto;
}

.add-reminder-modal-btn-text {
  font-size: 1.4rem;
  padding: 0;
  height: 2.4rem;
  width: 2.4rem;
  line-height: 2rem;
  display: inline-block;
}

.card {
  background-color: var(--light-gray);
  padding: 1.2rem 1.6rem 1.9rem;
  margin: 0 auto 1.6rem;
  margin-top: 3.1rem;
  -moz-border-radius: 2px;
  -webkit-border-radius: 2px;
  border-radius: 2px;
  -moz-box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
  -webkit-box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
}

.profile-img-card {
  width: 96px;
  height: 96px;
  margin: 0 auto 0.6rem;
  display: block;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  border-radius: 50%;
}

.active-reminder {
  background-color: #684e91;
  color: white;
}

.table-hover tbody tr.active-reminder:hover {
  background-color: #684e91 !important;
  color: white !important;
}

.glow {
  color: white;
  -webkit-animation: glow 1s ease-in-out infinite alternate;
  -moz-animation: glow 1s ease-in-out infinite alternate;
  animation: glow 1s ease-in-out infinite alternate;
}

@keyframes glow {
  from {
    text-shadow: 0 0 10px #fff, 0 0 20px #fff, 0 0 30px #6ca80c, 0 0 40px #6ca80c, 0 0 50px #6ca80c, 0 0 60px #6ca80c, 0 0 70px #6ca80c;
  }

  to {
    text-shadow: 0 0 20px #fff, 0 0 30px #8ee600, 0 0 40px #8ee600, 0 0 50px #8ee600, 0 0 60px #8ee600, 0 0 70px #8ee600, 0 0 80px #8ee600;
  }
}

.new-reminder-form {
  padding: 0;
  margin-top: 0;
  margin-bottom: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
}
