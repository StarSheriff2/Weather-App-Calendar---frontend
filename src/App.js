import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter as Router, Switch, Route, Link,
} from 'react-router-dom';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

import Login from './components/Login';
import Register from './components/Register';
import Profile from './components/Profile';
import Reminders from './components/Reminders';

import { logout } from './slices/auth';

import AuthVerify from './common/AuthVerify';

const App = () => {
  const { user: currentUser } = useSelector((state) => state.auth);
  const dispatch = useDispatch();

  const logOut = () => {
    dispatch(logout());
  };

  return (
    <Router>
      <div>
        <nav className="navbar navbar-expand navbar-dark bg-dark">
          <Link to="/" className="navbar-brand">
            WAC
          </Link>
          <div className="navbar-nav mr-auto">
            {currentUser && (
              <>
                <li className="nav-item">
                  <Link to="/home" className="nav-link">
                    Home
                  </Link>
                </li>
                <li className="nav-item">
                  <Link to="/reminders" className="nav-link">
                    Reminders
                  </Link>
                </li>
              </>
            )}
          </div>

          {currentUser && (
          // <div className="dropdown d-md-none">
          //   <button className="nav-button-mobile mx-0 px-1 py-3 my-2 bg-transparent border-0" type="button" id="dropdownMenu"
          //     data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><i className="px-2 fas fa-bars"></i></button>
          //   <div className="dropdown-menu" aria-labelledby="dropdownMenu">
          //   <li className="nav-item">
          //      <Link to="/profile" className="nav-link">
          //        {currentUser.name}
          //      </Link>
          //    </li>
          //    <li className="nav-item">
          //      <a href="/login" className="nav-link" onClick={logOut}>
          //        LogOut
          //      </a>
          //    </li>
          //   </div>
          // </div>

          <div className="dropdown">
            <button className="btn btn-secondary" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
              <i className="fas fa-bars" />
            </button>
            <div className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuButton">
              {/* <a className="dropdown-item" href="#">Action</a>
              <a className="dropdown-item" href="#">Another action</a>
              <a className="dropdown-item" href="#">Something else here</a> */}
              <li className="nav-item dropdown-item">
                <Link to="/profile" className="nav-link">
                  {currentUser.name}
                </Link>
              </li>
              <li className="nav-item dropdown-item">
                <a href="/login" className="nav-link" onClick={logOut}>
                  LogOut
                </a>
              </li>
            </div>
          </div>

          // <div className="navbar-nav ml-auto">
          // <li className="nav-item">
          //   <Link to="/profile" className="nav-link">
          //     {currentUser.name}
          //   </Link>
          // </li>
          // <li className="nav-item">
          //   <a href="/login" className="nav-link" onClick={logOut}>
          //     LogOut
          //   </a>
          // </li>
          // </div>
          )}
        </nav>

        <div className="container mt-3 px-1">
          <Switch>
            <Route exact path={['/', '/home']} component={Login} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/profile" component={Profile} />
            <Route path="/reminders" component={Reminders} />
          </Switch>
        </div>

        <AuthVerify logOut={logOut} />
      </div>
    </Router>
  );
};

export default App;
