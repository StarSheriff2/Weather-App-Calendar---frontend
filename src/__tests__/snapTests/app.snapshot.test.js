import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../tests/test-utils';
import App from '../../App';

afterEach(cleanup);

describe('App', () => {
  test('Renders the App', () => {
    const app = renderWithRedux(<App />).toJSON();

    expect(app).toMatchSnapshot();
  });
});
