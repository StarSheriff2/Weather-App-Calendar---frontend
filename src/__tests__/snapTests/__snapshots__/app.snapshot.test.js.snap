// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders the App 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      WAC
    </a>
    <div
      className="navbar-nav mr-auto"
    />
  </nav>
  <div
    className="container main-container mt-3 px-1"
  >
    <div
      className="col-md-12 login-form"
    >
      <div
        className="card card-container"
      >
        <img
          alt="profile-img"
          className="profile-img-card"
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
        />
        <form
          action="#"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="form-control"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="form-control"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn btn-primary btn-block"
              disabled={false}
              type="submit"
            >
              <span>
                Login
              </span>
            </button>
          </div>
        </form>
      </div>
      <p
        className="text-center"
      >
        Don't have an account?
      </p>
      <p
        className="text-center"
      >
        Sign up
         
        <a
          href="/register"
          onClick={[Function]}
        >
          here
        </a>
      </p>
    </div>
  </div>
  <div />
</div>
`;
